heat_algo.cpp

```diff
@@ -0,0 +1,214 @@
+#include <iostream>
+#include <chrono>
+#include <cmath>
+#include <vector>
+#include <numeric>
+#include <ranges>
+#include <execution>
+#include <algorithm>
+
```

Comment:
```
using exe_pol = std::execution::par_unseq;
// using exe_pol = std::execution::par;
```

Reply:
std::execution:par isn't recognised as a type but I can use macros to detect when compiled with nvc++

#ifdef __NVCOMPILER // Check if the NVCC compiler is being used
#define exe_pol std::execution::par_unseq // Use for nvc++ when targeting gpu
#else
#define exe_pol std::execution::par
#endif